toolchain/sem_ir/inst_namer.cpp

```diff
@@ -153,6 +150,9 @@ auto InstNamer::GetScopeIdOffset(ScopeIdTypeEnum id_enum) const -> int {
       offset += sem_ir_->specific_interfaces().size();
       [[fallthrough]];
     case ScopeIdTypeEnum::For<SpecificInterfaceId>:
+      offset += sem_ir_->vtables().size();
+      [[fallthrough]];
+    case ScopeIdTypeEnum::For<VtableId>:
       // All type-specific scopes are offset by `FirstEntityScope`.
```

Comment:
This looks weird to me. We're using vtables.size for the specific interface? And for Vtable we're using a fixed offset with a comment about specific scopes?

Reply:
Yes, it looks weird, and I tried to explain the bug fix in the description, but probably not well enough.
See the comment before the switch, each type's offset excludes its own type.
The bug was introduced in #5891, take a look at the code before the bug was introduced:
https://github.com/carbon-language/carbon-lang/blob/64139e5d65c855f59de63267810d8506c5a6aead/toolchain/sem_ir/inst_namer.cpp#L129-L155)

You can see each type adds the size for next switch case type.
The `[[fallthrough]]` and the comment above the switch is the key here I think.

This change fixes the code for that original logic.

Does this make sense or do I miss something in your comment?